Build fixes with gcc-16

**Describe the bug**
oe-core is soon bringing gcc-16 which results in edk2 recipes failing to build.

```
StringFuncs.c: In function ‘SplitStringByWhitespace’:
StringFuncs.c:113:15: error: variable ‘Item’ set but not used [-Werror=unused-but-set-variable=]
  # 113 |   UINTN       Item;
      |               ^~~~
cc1: all warnings being treated as errors
make[2]: *** [../Makefiles/footer.makefile:21: StringFuncs.o] Error 1
```

**To Reproduce**

apply gcc-16 patches to oe-core and then
bitbake standalone-mm-optee-tegra

**Additional context**
Please backport https://github.com/tianocore/edk2/pull/12079 this should fix it.

--- a/recipes-bsp/uefi/edk2-firmware-tegra-36.5.0.inc
+++ b/recipes-bsp/uefi/edk2-firmware-tegra-36.5.0.inc
@@ -38,7 +38,13 @@ SRC_URI += "\
     file://0004-fix-XusbControllerDxe-use-BaseMemoryLib-functions-14.patch;patchdir=../edk2-nvidia \
     file://0005-fix-L4TLauncher-allow-for-empty-missing-APPEND-line-.patch;patchdir=../edk2-nvidia \
     file://0006-PrePi-use-GCC5-instead-of-GCC-in-build-options.patch;patchdir=../edk2-nvidia \
-    file://0007-BaseTools-EfiRom-fix-compiler-warning.patch \
+    file://0007-BaseTools-StringFuncs-fix-gcc-16-warning.patch \
+    file://0008-BaseTools-EfiRom-fix-compiler-warning.patch \
+    file://0009-MdeModulePkg-CustomizedDisplayLib-fix-gcc-16-warning.patch \
+    file://0010-MdeModulePkg-SmbiosMeasurementDxe-fix-gcc-16-warning.patch \
+    file://0011-MdeModulePkg-DisplayEngineDxe-fix-gcc-16-warning.patch \
+    file://0012-MdeModulePkg-VarCheckHiiLib-fix-gcc-16-warning.patch \
+    file://0013-MdeModulePkg-CapsuleOnDiskLoadPei-fix-gcc-16-warning.patch \
 "
 
 S = "${UNPACKDIR}/edk2-tegra/edk2"

--- a/recipes-bsp/uefi/files/0007-BaseTools-StringFuncs-fix-gcc-16-warning.patch
+++ b/recipes-bsp/uefi/files/0007-BaseTools-StringFuncs-fix-gcc-16-warning.patch
@@ -1,0 +1,44 @@
+From 02d151049c3aff041680b59999d63b82f2d4c2df Mon Sep 17 00:00:00 2001
+From: Gerd Hoffmann <kraxel@redhat.com>
+Date: Thu, 29 Jan 2026 09:23:32 +0100
+Subject: [PATCH 07/13] BaseTools/StringFuncs: fix gcc 16 warning
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+StringFuncs.c: In function ‘SplitStringByWhitespace’:
+StringFuncs.c:113:15: error: variable ‘Item’ set but not used [-Werror=unused-but-set-variable=]
+  113 |   UINTN       Item;
+      |               ^~~~
+
+Upstream-Status: Backport [https://github.com/tianocore/edk2/pull/12047]
+Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
+Signed-off-by: Matt Madison <matt@madison.systems>
+---
+ BaseTools/Source/C/Common/StringFuncs.c | 3 +--
+ 1 file changed, 1 insertion(+), 2 deletions(-)
+
+diff --git a/BaseTools/Source/C/Common/StringFuncs.c b/BaseTools/Source/C/Common/StringFuncs.c
+index 53e44365e9..df02d9c808 100644
+--- a/BaseTools/Source/C/Common/StringFuncs.c
++++ b/BaseTools/Source/C/Common/StringFuncs.c
+@@ -110,7 +110,6 @@ SplitStringByWhitespace (
+   CHAR8       *EndOfSubString;
+   CHAR8       *EndOfString;
+   STRING_LIST *Output;
+-  UINTN       Item;
+ 
+   String = CloneString (String);
+   if (String == NULL) {
+@@ -120,7 +119,7 @@ SplitStringByWhitespace (
+ 
+   Output = NewStringList ();
+ 
+-  for (Pos = String, Item = 0; Pos < EndOfString; Item++) {
++  for (Pos = String; Pos < EndOfString;) {
+     while (isspace ((int)*Pos)) {
+       Pos++;
+     }
+-- 
+2.43.0
+

--- a/recipes-bsp/uefi/files/0008-BaseTools-EfiRom-fix-compiler-warning.patch
+++ b/recipes-bsp/uefi/files/0008-BaseTools-EfiRom-fix-compiler-warning.patch
@@ -1,7 +1,7 @@
-From 9af06ef3cbb052b142f9660c2c01e7aeb401300c Mon Sep 17 00:00:00 2001
+From 8658ae0ff154a4791a0501d8ae8b6ddbc18dd272 Mon Sep 17 00:00:00 2001
 From: Gerd Hoffmann <kraxel@redhat.com>
 Date: Mon, 8 Dec 2025 10:28:50 +0100
-Subject: [PATCH] BaseTools/EfiRom: fix compiler warning
+Subject: [PATCH 08/13] BaseTools/EfiRom: fix compiler warning
 MIME-Version: 1.0
 Content-Type: text/plain; charset=UTF-8
 Content-Transfer-Encoding: 8bit
@@ -13,8 +13,8 @@ EfiRom.c:78:17: error: assignment discards ‘const’ qualifier from pointer ta
 
 The assigned value is not used, so fix the warning by just removing it.
 
+Upstream-Status: Backport [https://github.com/tianocore/edk2/pull/12047]
 Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
-Upstream-Status: Backport [6e903b3504]
 Signed-off-by: Matt Madison <matt@madison.systems>
 ---
  BaseTools/Source/C/EfiRom/EfiRom.c | 3 +--

--- a/recipes-bsp/uefi/files/0009-MdeModulePkg-CustomizedDisplayLib-fix-gcc-16-warning.patch
+++ b/recipes-bsp/uefi/files/0009-MdeModulePkg-CustomizedDisplayLib-fix-gcc-16-warning.patch
@@ -1,0 +1,46 @@
+From 9a31e3f7c107bc928dd174b2073abb94defa705b Mon Sep 17 00:00:00 2001
+From: Gerd Hoffmann <kraxel@redhat.com>
+Date: Thu, 29 Jan 2026 09:32:15 +0100
+Subject: [PATCH 09/13] MdeModulePkg/CustomizedDisplayLib: fix gcc 16 warning
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+MdeModulePkg/Library/CustomizedDisplayLib/CustomizedDisplayLib.c: In function ‘CreateDialog’:
+MdeModulePkg/Library/CustomizedDisplayLib/CustomizedDisplayLib.c:435:18: error: variable ‘Count’ set but not used [-Werror=unused-but-set-variable=]
+  435 |   UINTN          Count;
+      |                  ^~~~~
+
+Upstream-Status: Backport [https://github.com/tianocore/edk2/pull/12047]
+Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
+Signed-off-by: Matt Madison <matt@madison.systems>
+---
+ .../Library/CustomizedDisplayLib/CustomizedDisplayLib.c       | 4 +---
+ 1 file changed, 1 insertion(+), 3 deletions(-)
+
+diff --git a/MdeModulePkg/Library/CustomizedDisplayLib/CustomizedDisplayLib.c b/MdeModulePkg/Library/CustomizedDisplayLib/CustomizedDisplayLib.c
+index b3da13d0a3..9efdd69430 100644
+--- a/MdeModulePkg/Library/CustomizedDisplayLib/CustomizedDisplayLib.c
++++ b/MdeModulePkg/Library/CustomizedDisplayLib/CustomizedDisplayLib.c
+@@ -432,7 +432,6 @@ CreateDialog (
+   UINTN          LargestString;
+   UINTN          LineNum;
+   UINTN          Index;
+-  UINTN          Count;
+   CHAR16         Character;
+   UINTN          Start;
+   UINTN          End;
+@@ -503,9 +502,8 @@ CreateDialog (
+   PrintCharAt ((UINTN)-1, (UINTN)-1, Character);
+   Character = BOXDRAW_VERTICAL;
+ 
+-  Count = 0;
+   VA_START (Marker, Key);
+-  for (Index = Top; Index + 2 < Bottom; Index++, Count++) {
++  for (Index = Top; Index + 2 < Bottom; Index++) {
+     String = VA_ARG (Marker, CHAR16 *);
+ 
+     if (String[0] == CHAR_NULL) {
+-- 
+2.43.0
+

--- a/recipes-bsp/uefi/files/0010-MdeModulePkg-SmbiosMeasurementDxe-fix-gcc-16-warning.patch
+++ b/recipes-bsp/uefi/files/0010-MdeModulePkg-SmbiosMeasurementDxe-fix-gcc-16-warning.patch
@@ -1,0 +1,58 @@
+From 0efe4dcd04e979915c920e29a95918ed25cd97b7 Mon Sep 17 00:00:00 2001
+From: Gerd Hoffmann <kraxel@redhat.com>
+Date: Thu, 29 Jan 2026 09:35:46 +0100
+Subject: [PATCH 10/13] MdeModulePkg/SmbiosMeasurementDxe: fix gcc 16 warning
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c: In function ‘GetSmbiosStringById’:
+MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c:221:10: error: variable ‘Size’ set but not used [-Werror=unused-but-set-variable=]
+  221 |   UINTN  Size;
+      |          ^~~~
+
+Upstream-Status: Backport [https://github.com/tianocore/edk2/pull/12047]
+Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
+Signed-off-by: Matt Madison <matt@madison.systems>
+---
+ .../Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c     | 4 ----
+ 1 file changed, 4 deletions(-)
+
+diff --git a/MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c b/MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c
+index d61edc846b..f9980e6be3 100644
+--- a/MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c
++++ b/MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c
+@@ -218,14 +218,12 @@ GetSmbiosStringById (
+   OUT  UINTN                    *StringLen
+   )
+ {
+-  UINTN  Size;
+   UINTN  StrLen;
+   CHAR8  *CharInStr;
+   UINTN  StringsNumber;
+   CHAR8  *String;
+ 
+   CharInStr     = (CHAR8 *)Head + Head->Length;
+-  Size          = Head->Length;
+   StringsNumber = 0;
+   StrLen        = 0;
+   //
+@@ -234,7 +232,6 @@ GetSmbiosStringById (
+   String = NULL;
+   while (*CharInStr != 0 || *(CharInStr+1) != 0) {
+     if (*CharInStr == 0) {
+-      Size += 1;
+       CharInStr++;
+     }
+ 
+@@ -256,7 +253,6 @@ GetSmbiosStringById (
+     // forward the pointer
+     //
+     CharInStr     += StrLen;
+-    Size          += StrLen;
+     StringsNumber += 1;
+     if (StringsNumber == StringId) {
+       break;
+-- 
+2.43.0
+

--- a/recipes-bsp/uefi/files/0011-MdeModulePkg-DisplayEngineDxe-fix-gcc-16-warning.patch
+++ b/recipes-bsp/uefi/files/0011-MdeModulePkg-DisplayEngineDxe-fix-gcc-16-warning.patch
@@ -1,0 +1,45 @@
+From a5cfe467eb7b500539b393b508f99a0dc8bc273c Mon Sep 17 00:00:00 2001
+From: Gerd Hoffmann <kraxel@redhat.com>
+Date: Thu, 29 Jan 2026 09:38:27 +0100
+Subject: [PATCH 11/13] MdeModulePkg/DisplayEngineDxe: fix gcc 16 warning
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+MdeModulePkg/Universal/DisplayEngineDxe/ProcessOptions.c: In function ‘CreateSharedPopUp’:
+MdeModulePkg/Universal/DisplayEngineDxe/ProcessOptions.c:590:11: error: variable ‘Count’ set but not used [-Werror=unused-but-set-variable=]
+  590 |   UINTN   Count;
+      |           ^~~~~
+
+Upstream-Status: Backport [https://github.com/tianocore/edk2/pull/12047]
+Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
+Signed-off-by: Matt Madison <matt@madison.systems>
+---
+ MdeModulePkg/Universal/DisplayEngineDxe/ProcessOptions.c | 4 +---
+ 1 file changed, 1 insertion(+), 3 deletions(-)
+
+diff --git a/MdeModulePkg/Universal/DisplayEngineDxe/ProcessOptions.c b/MdeModulePkg/Universal/DisplayEngineDxe/ProcessOptions.c
+index 0d4cfa4cf0..ae5891d691 100644
+--- a/MdeModulePkg/Universal/DisplayEngineDxe/ProcessOptions.c
++++ b/MdeModulePkg/Universal/DisplayEngineDxe/ProcessOptions.c
+@@ -587,7 +587,6 @@ CreateSharedPopUp (
+   )
+ {
+   UINTN   Index;
+-  UINTN   Count;
+   CHAR16  Character;
+   UINTN   Start;
+   UINTN   End;
+@@ -627,8 +626,7 @@ CreateSharedPopUp (
+   PrintCharAt ((UINTN)-1, (UINTN)-1, Character);
+   Character = BOXDRAW_VERTICAL;
+ 
+-  Count = 0;
+-  for (Index = Top; Index + 2 < Bottom; Index++, Count++) {
++  for (Index = Top; Index + 2 < Bottom; Index++) {
+     String = VA_ARG (Marker, CHAR16 *);
+ 
+     //
+-- 
+2.43.0
+

--- a/recipes-bsp/uefi/files/0012-MdeModulePkg-VarCheckHiiLib-fix-gcc-16-warning.patch
+++ b/recipes-bsp/uefi/files/0012-MdeModulePkg-VarCheckHiiLib-fix-gcc-16-warning.patch
@@ -1,0 +1,44 @@
+From ec3f8338ea42d94f551c46f685f230109ae3ceca Mon Sep 17 00:00:00 2001
+From: Gerd Hoffmann <kraxel@redhat.com>
+Date: Thu, 29 Jan 2026 10:34:52 +0100
+Subject: [PATCH 12/13] MdeModulePkg/VarCheckHiiLib: fix gcc 16 warning
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+MdeModulePkg/Library/VarCheckHiiLib/VarCheckHiiGenFromFv.c: In function ‘ParseFv’:
+MdeModulePkg/Library/VarCheckHiiLib/VarCheckHiiGenFromFv.c:263:34: error: variable ‘FfsIndex’ set but not used [-Werror=unused-but-set-variable=]
+  263 |   UINTN                          FfsIndex;
+      |                                  ^~~~~~~~
+
+Upstream-Status: Backport [https://github.com/tianocore/edk2/pull/12047]
+Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
+Signed-off-by: Matt Madison <matt@madison.systems>
+---
+ MdeModulePkg/Library/VarCheckHiiLib/VarCheckHiiGenFromFv.c | 3 +--
+ 1 file changed, 1 insertion(+), 2 deletions(-)
+
+diff --git a/MdeModulePkg/Library/VarCheckHiiLib/VarCheckHiiGenFromFv.c b/MdeModulePkg/Library/VarCheckHiiLib/VarCheckHiiGenFromFv.c
+index 562f5a02f6..4363eda115 100644
+--- a/MdeModulePkg/Library/VarCheckHiiLib/VarCheckHiiGenFromFv.c
++++ b/MdeModulePkg/Library/VarCheckHiiLib/VarCheckHiiGenFromFv.c
+@@ -260,7 +260,6 @@ ParseFv (
+   EFI_GUID                       NameGuid;
+   EFI_FV_FILE_ATTRIBUTES         FileAttributes;
+   UINTN                          Size;
+-  UINTN                          FfsIndex;
+   VAR_CHECK_VFR_DRIVER_INFO      *VfrDriverInfo;
+   LIST_ENTRY                     *VfrDriverLink;
+ 
+@@ -315,7 +314,7 @@ ParseFv (
+       Key = InternalVarCheckAllocateZeroPool (Fv2->KeySize);
+       ASSERT (Key != NULL);
+ 
+-      for (FfsIndex = 0; ; FfsIndex++) {
++      while (TRUE) {
+         FileType = EFI_FV_FILETYPE_ALL;
+         Status   = Fv2->GetNextFile (
+                           Fv2,
+-- 
+2.43.0
+

--- a/recipes-bsp/uefi/files/0013-MdeModulePkg-CapsuleOnDiskLoadPei-fix-gcc-16-warning.patch
+++ b/recipes-bsp/uefi/files/0013-MdeModulePkg-CapsuleOnDiskLoadPei-fix-gcc-16-warning.patch
@@ -1,0 +1,50 @@
+From aa679fa14f21eee7393fec47ebf267cf2c211e80 Mon Sep 17 00:00:00 2001
+From: Gerd Hoffmann <kraxel@redhat.com>
+Date: Thu, 29 Jan 2026 10:45:28 +0100
+Subject: [PATCH 13/13] MdeModulePkg/CapsuleOnDiskLoadPei: fix gcc 16 warning
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+MdeModulePkg/Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c:176:11: error: variable ‘Index’ set but not used [-Werror=unused-but-set-variable=]
+  176 |   UINTN   Index;
+      |           ^~~~~
+
+Upstream-Status: Backport [https://github.com/tianocore/edk2/pull/12047]
+Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
+Signed-off-by: Matt Madison <matt@madison.systems>
+---
+ .../Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c      | 3 ---
+ 1 file changed, 3 deletions(-)
+
+diff --git a/MdeModulePkg/Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c b/MdeModulePkg/Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c
+index 0ea4fb38f9..1bbf38afc4 100644
+--- a/MdeModulePkg/Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c
++++ b/MdeModulePkg/Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c
+@@ -173,7 +173,6 @@ RetrieveRelocatedCapsule (
+   IN UINTN  RelocCapsuleTotalSize
+   )
+ {
+-  UINTN   Index;
+   UINT8   *CapsuleDataBufEnd;
+   UINT8   *CapsulePtr;
+   UINT32  CapsuleSize;
+@@ -244,7 +243,6 @@ RetrieveRelocatedCapsule (
+   // Re-iterate the capsule buffer to create Capsule hob & Capsule Name Str Hob for each Capsule saved in relocated capsule file
+   //
+   CapsulePtr = RelocCapsuleBuf;
+-  Index      = 0;
+   while (CapsulePtr < CapsuleDataBufEnd) {
+     CapsuleSize = ((EFI_CAPSULE_HEADER *)CapsulePtr)->CapsuleImageSize;
+     BuildCvHob ((EFI_PHYSICAL_ADDRESS)(UINTN)CapsulePtr, CapsuleSize);
+@@ -252,7 +250,6 @@ RetrieveRelocatedCapsule (
+     DEBUG ((DEBUG_INFO, "Capsule saved in address %x size %x\n", CapsulePtr, CapsuleSize));
+ 
+     CapsulePtr += CapsuleSize;
+-    Index++;
+   }
+ 
+   return EFI_SUCCESS;
+-- 
+2.43.0
+

